### ISSUE TITLE
Sort out web font loading

### ISSUE THREAD

**Issue by i-like-robots** [Financial-Times/dotcom-page-kit#763]
The implementation of web fonts is currently split between the shell and layout components and is comprised of several parts:

1. Resource hints for the required font files (shell)
2. The progressive loading class names (layout)
3. The embedded JS snippet which checks which cookies are set (layout)
4. The initialisation of the `o-typography` component (layout)
5. The CSS font face rules (n-ui-foundations)

It is not ideal having the implementation split between so many components. The integration is also not accessible from apps so there is no way to indicate which fonts are needed and which aren't.

Most importantly, the embedded JS snippet should be placed in the document `<head>` to avoid [FOUT](https://css-tricks.com/fout-foit-foft/) but since the `<Shell />` component does not provide a client-side JS module that would mean initialising `o-typography` separately elsewhere 😬 

In short we need to solve the following problems:

1. Colocate all web font configuration and code
2. Move the embedded JS snippet to the most optimal location
3. Ensure additional web fonts (such as Metric Medium on article pages) can be configured

## References

- https://github.com/Financial-Times/o-typography#setup-progressive-loading
- https://www.zachleat.com/web/the-compromise/
- https://www.zachleat.com/web/comprehensive-webfonts/
- https://andydavies.me/blog/2019/02/12/preloading-fonts-and-the-puzzle-of-priorities/
- https://tech.ebayinc.com/engineering/ebays-font-loading-strategy/

**Comment by i-like-robots** [comment]
I have scheduled a chat with @notlee to discuss web font loading for 2020.

**Comment by i-like-robots** [comment]
We currently configure 5 fonts in our CSS https://github.com/Financial-Times/n-ui-foundations/blob/master/typography/main.scss#L5-L15

But only 4 are configured to be loaded by `o-typography`: https://github.com/Financial-Times/o-typography/blob/ce625323d492395784f384358a6e65d0c43df484/src/js/typography.js#L25-L46

**Comment by notlee** [comment]
Hey, `o-typography` js doesn't load fonts. It stores which fonts have loaded (in a cookie 😷) and switches some class names so we can resize fallback fonts, to minimise the effect of FOUT. We don't resize all fallback fonts faces, only four, but may use more

**Comment by i-like-robots** [comment]
Have replaced "loaded" with "watched" =]

**Comment by i-like-robots** [comment]
I'm going to close this as #770 has centralised all of the font loading configuration and information.
